<launch>

    <arg name="marker0Id"         default="100"/>
    <arg name="marker1Id"         default="101"/>
    <arg name="marker2Id"         default="102"/>
    <arg name="marker3Id"         default="103"/>
    <arg name="marker4Id"         default="104"/>
    <arg name="markerSize"        default="0.058"/> <!-- in m -->
    <arg name="eye"               default="left"/>
    <arg name="marker0_frame"     default="aruco_marker0_frame"/>
    <arg name="marker1_frame"     default="aruco_marker1_frame"/>
    <arg name="marker2_frame"     default="aruco_marker2_frame"/>
    <arg name="marker3_frame"     default="aruco_marker3_frame"/>
    <arg name="marker4_frame"     default="aruco_marker4_frame"/>
    <arg name="ref_frame"         default="camera_frame"/>
    <arg name="camera_frame"      default="camera_frame"/>
    <arg name="dct_normalization" default="True" />
    <arg name="dct_filter_size"   default="2" />
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->


    <node pkg="aruco_ros" type="3dgrid" name="aruco_3dgrid" output="screen">    
        <remap from="/camera_info" to="/mv_29700299/camera_info" /> <!-- Remap this to camera topics -->
    	<remap from="/image" to="/mv_29700299/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <!--<param name="markers_id"         value="[$(arg marker0Id), $(arg marker1Id), $(arg marker2Id), $(arg marker3Id), $(arg marker4Id)]"/>-->
        <param name="marker0_id"         value="$(arg marker0Id)"/>
        <param name="marker1_id"         value="$(arg marker1Id)"/>
        <param name="marker2_id"         value="$(arg marker2Id)"/>
        <param name="marker3_id"         value="$(arg marker3Id)"/>
        <param name="marker4_id"         value="$(arg marker4Id)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered to -->
        <param name="camera_frame"       value="$(arg camera_frame)"/>
        <param name="marker0_frame"      value="$(arg marker0_frame)" />
        <param name="marker1_frame"      value="$(arg marker1_frame)" />
        <param name="marker2_frame"      value="$(arg marker2_frame)" />
        <param name="marker3_frame"      value="$(arg marker3_frame)" />
        <param name="marker4_frame"      value="$(arg marker4_frame)" />
        <param name="normalizeImage"     value="$(arg dct_normalization)"/>
        <param name="dct_components_to_remove" value="$(arg dct_filter_size)"/>
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>


</launch>